crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1408,28 +1412,43 @@ impl<'db> TypeInferenceBuilder<'db> {
         // If the target defines, e.g., `__iadd__`, infer the augmented assignment as a call to that
         // dunder.
         if let Type::Instance(class) = target_type {
-            let class_member = class.class_member(self.db, op.in_place_dunder());
-            if !class_member.is_unbound() {
-                let call = class_member.call(self.db, &[target_type, value_type]);
-                return match call.return_ty_result(
-                    self.db,
-                    AnyNodeRef::StmtAugAssign(assignment),
-                    &mut self.diagnostics,
-                ) {
-                    Ok(t) => t,
-                    Err(e) => {
+            match class.class_member(self.db, op.in_place_dunder()) {
+                SymbolLookupResult::Unbound => {}
+                SymbolLookupResult::Type(class_member, boundedness) => {
+                    if boundedness == Boundedness::MayBeUnbound {
                         self.diagnostics.add(
                             assignment.into(),
-                            "unsupported-operator",
+                            "call-potentially-unbound-method",
                             format_args!(
-                                "Operator `{op}=` is unsupported between objects of type `{}` and `{}`",
-                                target_type.display(self.db),
-                                value_type.display(self.db)
+                                "Call to potentially unbound method `{}` on object of type `{}`",
+                                op.in_place_dunder(),
+                                target_type.display(self.db)
                             ),
                         );
-                        e.return_ty()
                     }
```

Comment:
I don't think a `call-potentially-unbound-method` diagnostic is the desired behavior here; I think we should be falling back to the binary-op behavior below and unioning that with the result of calling the maybe-bound method. If binary-op is not supported then we would raise a diagnostic. I think @charliermarsh was planning to pick that up once this lands.